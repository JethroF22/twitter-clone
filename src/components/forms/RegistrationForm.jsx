import React, { Component } from 'react';

class RegistrationForm extends Component {
  state = {
    username: '',
    password: '',
    confirmPassword: '',
    handle: '',
    email: ''
  };

  render() {
    return (
      <div>
        <h3>Create an account</h3>
        <form onSubmit={() => console.log('submitted')}>
          <label htmlFor="username">Username: </label>
          <input
            type="text"
            name="username"
            value={this.state.username}
            onChange={() => console.log('Username changed')}
          />
          <br />
          <label htmlFor="email">Email: </label>
          <input
            type="email"
            name="email"
            value={this.state.email}
            onChange={() => console.log('Email changed')}
          />
          <br />
          <label htmlFor="handle">Handle: </label>
          <input
            type="text"
            name="handle"
            value={this.state.handle}
            onChange={() => console.log('Handle changed')}
            placeholder="Example: @i_am_jerry"
          />
          <br />
          <label htmlFor="password">Password: </label>
          <input
            type="password"
            name="password"
            value={this.state.password}
            onChange={() => console.log('Password changed')}
          />
          <br />
          <label htmlFor="confirmPassword">Confirm Password: </label>
          <input
            type="password"
            name="confirmPassword"
            value={this.state.confirmPassword}
            onChange={() => console.log('Confirm Password changed')}
          />
          <br />
        </form>
      </div>
    );
  }
}

export default RegistrationForm;
