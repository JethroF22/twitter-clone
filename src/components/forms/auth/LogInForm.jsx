import React, { Component } from 'react';
import { connect } from 'react-redux';
import _ from 'lodash';
import validator from 'validator';

import { logIn } from '../../../actions/auth';
import {
  actionStatusMessages,
  errorMessages
} from '../../../../config/const.json';

const {
  SUCCESS_MESSAGE,
  FAILED_MESSAGE,
  IN_PROGRESS_MESSAGE
} = actionStatusMessages;
const { INVALID_CREDENTIALS } = errorMessages;

export class LogInForm extends Component {
  state = {
    email: '',
    password: '',
    errors: {}
  };

  onChange = e => {
    const { value, name } = e.target;
    this.setState(() => ({
      [name]: value,
      errors: {}
    }));
  };

  onSubmit = e => {
    e.preventDefault();

    const credentials = _.pick(this.state, ['email', 'password']);

    if (validator.isEmail(credentials.email)) {
      this.props.logIn(credentials).then(() => {
        if (this.props.actionStatus === SUCCESS_MESSAGE) {
          alert('successful');
        } else if (this.props.actionStatus === FAILED_MESSAGE) {
          this.setState(() => ({
            errors: {
              authenticationError: INVALID_CREDENTIALS
            }
          }));
        }
      });
    } else {
      this.setState(() => ({
        errors: {
          email: 'Invalid email address'
        }
      }));
    }
  };

  render() {
    return (
      <div>
        <h3>Log In</h3>
        <form onSubmit={this.onSubmit}>
          {this.state.errors.authenticationError && (
            <p>{this.state.errors.authenticationError}</p>
          )}
          {this.state.errors.email && <p>{this.state.errors.email}</p>}
          <label htmlFor="email">Email: </label>
          <input
            type="email"
            name="email"
            value={this.state.email}
            onChange={this.onChange}
          />
          <br />
          <label htmlFor="password">Password: </label>
          <input
            type="password"
            name="password"
            value={this.state.password}
            onChange={this.onChange}
          />
          <br />
          <button
            type="submit"
            disabled={this.props.actionStatus === IN_PROGRESS_MESSAGE}
          >
            Log In
          </button>
        </form>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  actionStatus: state.status.actionStatus,
  errorType: state.error.errorType,
  errorMessage: state.error.errorMessage
});

const mapDispatchToProps = dispatch => ({
  logIn: credentials => dispatch(logIn(credentials))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(LogInForm);
