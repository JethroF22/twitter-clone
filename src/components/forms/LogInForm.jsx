import React, { Component } from 'react';
import { connect } from 'react-redux';
import _ from 'lodash';

import { logIn } from '../../actions/auth';

class LogInForm extends Component {
  state = {
    email: '',
    password: '',
    errors: {}
  };

  render() {
    return (
      <div>
        <h3>Log In</h3>
        <form>
          {this.state.errors.authenticationError && (
            <p>{this.state.errors.authenticationError}</p>
          )}
          <label htmlFor="email">Email: </label>
          <input
            type="email"
            name="email"
            value={this.state.email}
            onChange={() => console.log('Email changed')}
          />
          <br />
          <label htmlFor="password">Password: </label>
          <input
            type="password"
            name="password"
            value={this.state.password}
            onChange={() => console.log('Password changed')}
          />
          <br />
          <button type="submit">Log In</button>
        </form>
      </div>
    );
  }
}

export default LogInForm;
