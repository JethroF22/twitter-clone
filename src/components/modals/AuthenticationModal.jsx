import React from 'react';
import Modal from 'react-modal';

import RegistrationForm from '../forms/RegistrationForm';
import LogInForm from '../forms/LogInForm';

const AuthenticationModal = props => (
  <Modal
    isOpen={props.modalIsOpen}
    contentLabel={props.modalContent}
    onRequestClose={props.closeModal}
    closeTimeoutMS={200}
  >
    <button onClick={props.closeModal}>&times;</button>
    {props.modalContent === 'Sign Up' ? <RegistrationForm /> : <LogInForm />}
  </Modal>
);

export default AuthenticationModal;
