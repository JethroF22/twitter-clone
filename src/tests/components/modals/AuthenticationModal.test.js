import React from 'react';
import { shallow } from 'enzyme';

import AuthenticationModal from '../../../components/modals/AuthenticationModal';
import LogInForm from '../../../components/forms/auth/LogInForm';
import RegistrationForm from '../../../components/forms/auth/RegistrationForm';

let wrapper, modalIsOpen, modalContent, triggerModal, closeModal;

describe('AuthenticationModal', () => {
  beforeEach(() => {
    triggerModal = jest.fn();
    closeModal = jest.fn();
    modalIsOpen = true;
    modalContent = 'Sign Up';
    wrapper = shallow(
      <AuthenticationModal
        triggerModal={triggerModal}
        closeModal={closeModal}
        modalIsOpen={modalIsOpen}
        modalContent={modalContent}
      />
    );
  });

  test('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should render the correct form', () => {
    expect(wrapper.containsMatchingElement(<RegistrationForm />)).toBe(true);
    wrapper.setProps({ modalContent: 'Log In' });
    expect(wrapper.containsMatchingElement(<LogInForm />)).toBe(true);
  });

  test('should handle closeModal', () => {
    wrapper.find('button[name="closeModal"]').simulate('click');
    expect(closeModal).toHaveBeenCalled();
  });
});
