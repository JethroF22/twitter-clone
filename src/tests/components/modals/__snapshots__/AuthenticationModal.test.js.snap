// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthenticationModal should render correctly 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  closeTimeoutMS={200}
  contentLabel="Sign Up"
  isOpen={true}
  onRequestClose={[MockFunction]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <button
    name="closeModal"
    onClick={[MockFunction]}
  >
    Ã—
  </button>
  <Connect(RegistrationForm) />
</Modal>
`;
